dialect/sql/builder.go

```diff
@@ -692,6 +710,27 @@ func (i *InsertBuilder) Query() (string, []interface{}) {
 			})
 		}
 	}
+
+	if len(i.conflictColumns) > 0 && i.postgres() {
+		i.WriteString(" ON CONFLICT ")
+		i.Nested(func(b *Builder) {
+			b.IdentComma(i.conflictColumns...)
+		})
+		i.WriteString(" DO UPDATE SET ")
+		for j, c := range i.updateColumns {
+			if j > 0 {
+				i.Comma()
+			}
+			i.Ident(c).WriteString(" = ")
+			switch v := i.updateValues[j].(type) {
+			case Querier:
+				i.Join(v)
+			default:
+				i.Arg(v)
```

Comment:
Instead of using arg you should use the dialect specific update syntax so that this will work with multiple rows.

Postgres
```
insert into segments(id, departure_hour) 
values 
    (1153, 2), 
    (1156, 4), 
    (1154, 2) 
on conflict do update set 
    departure_hour = excluded.departure_hour;
```

Mysql
https://stackoverflow.com/questions/2714587/mysql-on-duplicate-key-update-for-multiple-rows-insert-in-single-query

Not sure if sqlite has an equivalent

Reply:
Thanks for the input, I'll take a look at that, and I like the approach you took to supporting multiple dialects.